tests/components/homekit/test_accessories.py

```diff
@@ -130,7 +130,7 @@ async def test_battery_service(hass, hk_driver, caplog):
     await hass.async_block_till_done()
 
     acc = HomeAccessory(hass, hk_driver, "Battery Service", entity_id, 2, None)
-    acc.update_state = lambda x: None
+    acc.async_update_state = lambda x: None
```

Comment:
Lambdas will be scheduled on the executor thread pool if they are passed to `hass.async_add_job`.

Reply:
looks like they are being used for testing batteries not attached to an accessory since it will call `async_update_state` which was `update_state` which is expected to be implemented by the accessory.  I'll mock it out and verify its being called with the new state.